File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    for i, j in enumerate(prices):
        for k in range(i+1, len(prices)):
            if prices[k]-j<0:
                answer.append(k-i)
                break
        if len(answer) != i+1:
            answer.append(len(prices)-1-i)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> stack = new Stack<>();
       
        for (int i = 0; i<prices.length; i++) {
            while (true) {
                
                if (!stack.isEmpty() && prices[i] < prices[stack.peek()]) {
                    int top = stack.peek();
                    answer[top] = i-top;
                    stack.pop();
                } else break;
            }
            stack.push(i);
        }
        
        int top = stack.pop();
        while (!stack.isEmpty()) {
            int k = stack.pop();
            answer[k] = top - k;
        }
        
        return answer;
    }
}